java/code/src/com/redhat/rhn/domain/server/MinionServerFactory.java

```diff
@@ -61,11 +54,16 @@ public class MinionServerFactory extends HibernateFactory {
      * @return the Server found
      */
     public static List<MinionServer> lookupByOrg(Long orgId) {
-        return HibernateFactory.getSession()
-                .createCriteria(MinionServer.class)
-                .setResultTransformer(CriteriaSpecification.DISTINCT_ROOT_ENTITY)
-                .add(Restrictions.eq("org.id", orgId))
-                .list();
+        return getSession().createNativeQuery("""
+                                      SELECT DISTINCT s.*, r.*
+                                      FROM suseminioninfo s
+                                      JOIN rhnserver r ON s.server_id = r.id
+                                      WHERE
+                                      s.server_id IN
+                                      (SELECT id FROM rhnServer WHERE org_id = :org)
+                                      """, MinionServer.class)
```

Comment:
we had problems with "DISTINCT" in the query. Maybe we can test this query if it really report every minion only 1 time?

Reply:
do you mean:
- remove "DISTINCT" and check if unit test are still ok or
- add new unit tests for these particular queries
?

Reply:
Well, no. The queston is: does this "DISTINCT" here what we want?
Yesterday I found out that it was not doing what I expected.

Reply:
well, I tried to run the queries (with and without) distinct in our large environment: the result was the same. Morever we have in `suseMinionInfo`:
```
    server_id             NUMERIC NOT NULL
                              CONSTRAINT suse_minion_info_sid_fk
                                  REFERENCES rhnServer (id)
                                  ON DELETE CASCADE,
``` 
and
```
ALTER TABLE suseMinionInfo
 ADD CONSTRAINT rhn_minion_info_sid_uq UNIQUE (server_id);
```
So I think we cannot have duplicated here and distinct is useless. What do you think?

Reply:
Sounds good